packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -282,6 +286,15 @@ class MemoryBodyState extends State<MemoryBody>
       }
     });
 
+    addAutoDisposeListener(controller.androidCollectionEnabled, () {
+      setState(() {
+        if (!controller.androidCollectionEnabled.value &&
+            controller.isAndroidChartVisible) {
+          controller.toggleAndroidChartVisibility();
```

Comment:
what if the chart was previously not visible, and the checkbox was toggled to true? do we also need to toggle chart visibility in this case?

Reply:
This will enable/disable the Android Memory button if on.  I don't think to display a new chart when enabled makes sense.  However, when android collection is off I do hide the android chart - because it's useless if there is no data.

I'm not sure flipping on is necessary as the data is record and when clicked on the all collected data is available.